kythe/go/extractors/config/runextractor/mavencmd/pom_xml_modifier_test.go

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2018 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package mavencmd
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"kythe.io/kythe/go/test/testutil"
+)
+
+const testDataDir = "testdata"
+
+func TestPreProcess(t *testing.T) {
+	testcases := []struct {
+		inputFile          string
+		expectedOutputFile string
+	}{
+		{"modified-pom.xml", "modified-pom.xml"},
+		{"plain-pom.xml", "modified-pom.xml"},
+		{"other-pom.xml", "modified-pom.xml"},
+	}
+
+	for _, tcase := range testcases {
+		// Copy the file into a temp file.
+		tf, err := ioutil.TempFile("", tcase.inputFile)
+		if err != nil {
+			t.Fatalf("creating temp file: %v", err)
+		}
+		tfName := tf.Name()
+		defer os.Remove(tfName)
```

Comment:
The defers don't happen till the function returns. Which is fine, given there are only three, but you could also use a subtest for this if you like: https://blog.golang.org/subtests

```
t.Run(tcase.inputFile, func(t *testing.T) {
   …the code you have now…
})
```

Reply:
That also has the nice side benefit that a fatal in the test body doesn't prevent the rest of the cases in the loop from running (whereas, at present, that terminates the whole test).

Reply:
oh that's slick